great_expectations/checkpoint/checkpoint.py

```diff
@@ -310,6 +321,9 @@ def _run_validation(
     ) -> None:
         if validation_dict is None:
             validation_dict = {}
+            validation_dict["id_"] = substituted_runtime_config.get(
+                "default_validation_id"
+            )
```

Comment:
`validation_dict` is only ever `None` if no `validations` are present in the `run` signature (or in the `Checkpoint`'s internal state)